src/Evolution/Initialization/Tags.hpp

```diff
@@ -3,22 +3,80 @@
 
 #pragma once
 
+#include <array>
 #include <cmath>
 #include <string>
+#include <vector>
 
 #include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/Tags.hpp"
+#include "Options/Options.hpp"
 #include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
 namespace Initialization {
+namespace OptionTags {
+struct FuncOfTimeFromFile {
+  static constexpr OptionString help{
+      "Options for reading a FunctionOfTime from an H5 file"};
+};
+
+struct FuncOfTimeFile {
+  using type = std::string;
+  static constexpr OptionString help{"Path to an H5 Dat file"};
+  using group = FuncOfTimeFromFile;
+};
+
+struct FuncOfTimeSetNames {
+  using type = std::vector<std::array<std::string, 2>>;
+  static constexpr OptionString help{
+      "String pairs mapping datasets to arbitrary names"};
+  using group = FuncOfTimeFromFile;
+};
+}  // namespace OptionTags
+
 /// \ingroup InitializationGroup
 /// \brief %Tags used during initialization of parallel components.
 namespace Tags {
+/// \ingroup ControlSystemGroup
+/// \brief The path to an H5 Dat file containing function-of-time data to read.
+struct FuncOfTimeFile : db::SimpleTag {
+  static std::string name() noexcept { return "FuncOfTimeFile"; }
+  using type = std::string;
+  using option_tags =
+      tmpl::list<::Initialization::OptionTags::FuncOfTimeFile>;
+  template <typename Metavariables>
+  static std::string create_from_options(
+      const std::string& function_of_time_file) noexcept {
+    return function_of_time_file;
+  }
+};
+
+/// \ingroup ControlSystemGroup
+/// \brief List of pairs of strings mapping datasets to unordered-map keys
+///
+/// The first string in each pair names a dataset in an H5 file, and the
```

Comment:
I don't know what a `dataset in an H5 file` is. That's not a data format we use.

Reply:
I attempted to clarify...when you next take a look, please let me know if it makes sense this time